six/modules/c++/cphd/unittests/test_cphd_read_unscaled_int.cpp

```diff
@@ -0,0 +1,307 @@
+
+    if (scale)
+    {
+        for (size_t ii = 0; ii < scaleFactors.size(); ++ii)
+        {
+            scaleFactors[ii] *= 2;
+        }
+    }
+
+    return scaleFactors;
+}
+
+template<typename T>
+void writeCPHD(const std::string& outPathname, size_t numThreads,
+        const types::RowCol<size_t> dims,
+        const std::vector<std::complex<T> >& writeData)
+{
+    const size_t numChannels = 1;
+    const std::vector<size_t> numVectors(numChannels, dims.row);
+
+    cphd::Metadata metadata;
+
+    metadata.data.numCPHDChannels = numChannels;
+    for (size_t ii = 0; ii < metadata.data.numCPHDChannels; ++ii)
+    {
+        metadata.data.arraySize.push_back(
+                cphd::ArraySize(dims.row, dims.col));
+    }
+
+    //! Must set the sample type
+    if (sizeof(writeData[0]) == 2)
+    {
+        metadata.data.sampleType = cphd::SampleType::RE08I_IM08I;
+    }
+    else if (sizeof(writeData[0]) == 4)
+    {
+        metadata.data.sampleType = cphd::SampleType::RE16I_IM16I;
+    }
+    else if (sizeof(writeData[0]) == 8)
+    {
+        metadata.data.sampleType = cphd::SampleType::RE32F_IM32F;
+    }
+
+    //! We must have a radar mode set
+    metadata.collectionInformation.radarMode = cphd::RadarModeType::SPOTLIGHT;
+
+    //! We must have corners set
+    for (size_t ii = 0; ii < six::LatLonAltCorners::NUM_CORNERS; ++ii)
+    {
+        metadata.global.imageArea.acpCorners.getCorner(ii).setLat(0.0);
+        metadata.global.imageArea.acpCorners.getCorner(ii).setLon(0.0);
+        metadata.global.imageArea.acpCorners.getCorner(ii).setAlt(0.0);
+    }
+
+    //! Add a channel parameter. This will write without this but it will
+    //  not read.
+    cphd::ChannelParameters param;
+    metadata.channel.parameters.push_back(param);
+
+    //! We must set SRP Type
+    metadata.srp.srpType = cphd::SRPType::STEPPED;
+
+    //! We must set domain parameters
+    metadata.global.domainType = cphd::DomainType::FX;
+    metadata.vectorParameters.fxParameters.reset(new cphd::FxParameters());
+
+    cphd::VBM vbm(numChannels,
+                  numVectors,
+                  false,
+                  false,
+                  false,
+                  metadata.global.domainType);
+
+    cphd::CPHDWriter writer(metadata, numThreads);
+    writer.writeMetadata(outPathname, vbm);
+    for (size_t ii = 0; ii < numChannels; ++ii)
+    {
+        writer.writeCPHDData(&writeData[0], dims.area());
+    }
+}
+
+std::vector<std::complex<float> > checkData(const std::string& pathname,
+        size_t numThreads,
+        const std::vector<double>& scaleFactors,
+        bool scale,
+        const types::RowCol<size_t>& dims)
+{
+    cphd::CPHDReader reader(pathname, numThreads);
+    cphd::Wideband& wideband = reader.getWideband();
+    std::vector<std::complex<float> > readData(dims.area());
+
+    size_t sizeInBytes = readData.size() * sizeof(readData[0]);
+    sys::ubyte scratchData[sizeInBytes];
```

Comment:
Can't create an array on the stack whose size you don't know at compile time (gcc is being nice).  Use a vector or ScopedArray.